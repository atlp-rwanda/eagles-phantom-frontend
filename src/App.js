import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "./redux/actions/counter";
import './scss/styles.scss'

class App extends Component {
  render() {
    return (
      <div>
        <h1>Hello!! Welcome to phantom App.</h1>
        <h2>The following is the counter of number.</h2>
        <h3 id="value" aria-label="value" >value: {this.props.value}</h3>
        <button id="incrementButton" aria-label="incrementButton" name="increment" onClick={this.props.increment}>Increment</button>
        <button id="decrementButton" aria-label="decrementButton" name="decrement" onClick={this.props.decrement}>decrement</button>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    value: state.count.value,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    increment: () => dispatch(actions.increment()),
    decrement: () => dispatch(actions.decrement()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
