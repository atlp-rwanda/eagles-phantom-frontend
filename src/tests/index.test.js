import React from 'react'
import { render } from "@testing-library/react"
import App from '../components/App'


const AppComponent = () => {
  return render(<App />)
}
describe("APP", () => {
  it("should render the component", () => {
      const { asFragment } = AppComponent();
      expect(asFragment(<App />)).toMatchSnapshot()
  })
})