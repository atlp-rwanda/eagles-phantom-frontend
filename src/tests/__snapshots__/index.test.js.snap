// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should render the component 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hello!! Welcome to phantom App.
    </h1>
  </div>
</DocumentFragment>
`;
